src/olympia/promoted/models.py

```diff
@@ -21,9 +21,9 @@ def __getattr__(self, attribute):
             return self.values_list(attribute, flat=True)
         raise AttributeError(f'PromotedGroup has no attribute: {attribute}')
 
-    @property
-    def name(self):
-        return ', '.join(self.__getattr__('name'))
+    def get_names(self, needs_badged=True):
```

Comment:
NIT: I don't see the value in changing "badged" to "needs_badged" if they mean the same thing.

Reply:
TBH adding the argument here feels like an unecessary abstraction. The queryset already has the ability to filter by properties. Just use the filter and return the name property where needed.

LIke I've said above, if we wanted a convenient way to query promoted groups that are badged we could add that on the `promoted_groups` method but even that is probably overkill.